src/main/java/org/icgc/argo/program_service/grpc/ProgramServiceImpl.java

```diff
@@ -261,10 +263,66 @@ public void removeProgram(RemoveProgramRequest request, StreamObserver<Empty> re
     responseObserver.onCompleted();
   }
 
+  @Override
+  public void listCancers(Empty request, StreamObserver<ListCancersResponse> responseObserver) {
+    val cancers = programService.listCancers();
+    val listCancersResponse = programConverter.cancerEntitiesToListCancersResponse(cancers);
+    responseObserver.onNext(listCancersResponse);
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public void listPrimarySites(Empty request, StreamObserver<ListPrimarySitesResponse> responseObserver) {
+    val sites = programService.listPrimarySites();
+    val listPrimarySitesResponse = programConverter.primarySiteEntitiesToListPrimarySitesResponse(sites);
+    responseObserver.onNext(listPrimarySitesResponse);
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public void listCountries(Empty request, StreamObserver<ListCountriesResponse> responseObserver) {
+    val countries = programService.listCountries();
+    val listCountriesResponse = programConverter.countryEntitiesToListCountriesResponse(countries);
+    responseObserver.onNext(listCountriesResponse);
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public void listRegions(Empty request, StreamObserver<ListRegionsResponse> responseObserver) {
+    val regions = programService.listRegions();
+    val listRegionsResponse = programConverter.regionEntitiesToListRegionsResponse(regions);
+    responseObserver.onNext(listRegionsResponse);
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public void listInstitutions(Empty request, StreamObserver<ListInstitutionsResponse> responseObserver) {
+    val institutions = programService.listInstitutions();
+    val listInstitutionsResponse = programConverter.institutionEntitiesToListInstitutionsResponse(institutions);
+    responseObserver.onNext(listInstitutionsResponse);
+    responseObserver.onCompleted();
+  }
+
   private String getExceptionMessage(NestedRuntimeException e) {
     return e.getMostSpecificCause().getMessage();
   }
 
+  @Override
+  public void addInstitutions(AddInstitutionsRequest request, StreamObserver<AddInstitutionsResponse> responseObserver) {
+    val names = request.getNamesList().stream().map(name -> commonConverter.unboxStringValue(name)).collect(toImmutableList());
+    try {
+      val institutions = programService.addInstitutions(names);
+      val response = programConverter.institutionsToAddInstitutionsResponse(institutions);
+      responseObserver.onNext(response);
+      responseObserver.onCompleted();
+    } catch (DataIntegrityViolationException e){
+      log.error("Exception throw in addInstitutions: {}", e.getMessage());
+      e.printStackTrace();
```

Comment:
This prints to standard error stream. I think log.error(Message, Throwable) should be able to handle the stack trace.
Reference:
https://www.slf4j.org/api/src-html/org/slf4j/Logger.html#line.723